pkg/remoteimage/pull.go

```diff
@@ -40,14 +43,31 @@ func (p puller) ImageSize(ctx context.Context) int {
 }
 
 func (p puller) Pull(ctx context.Context) (err error) {
+	startTime := time.Now()
+	defer func() { // must capture final value of "err"
+		elapsed := time.Since(startTime).Seconds()
+		metrics.ImagePullTimeHist.WithLabelValues(metrics.BoolToString(err != nil)).Observe(elapsed)
+		metrics.ImagePullTime.WithLabelValues(p.image.String(), metrics.BoolToString(err != nil)).Set(elapsed)
+		go func() {
+			//TODO: this is a hack to ensure data is cleared in a reasonable timeframe and does not build up.
+			// pushgateway may remove the need for this. https://prometheus.io/docs/practices/pushing/
+			time.Sleep(1 * time.Minute)
+			metrics.ImagePullTime.DeleteLabelValues(p.image.String(), metrics.BoolToString(err != nil))
+		}()
```

Comment:
note the 1m cleanup logic spawned in separate go routine. this prevents pile-up of metrics for prometheus to scrape.